src/app/index.tsx

```diff
@@ -43,14 +43,28 @@ import { AppRoutes } from '@app/routes';
 import '@app/app.css';
 import { ServiceContext, defaultServices } from '@app/Shared/Services/Services';
 
-const App: React.FunctionComponent = () => (
+const App: React.FunctionComponent = () => {
+
+  const [authenticated, setAuthenticated] = React.useState(false);
+
+  const handleAuthenticated = React.useCallback(() => {
+    setAuthenticated(true);
+  }, [setAuthenticated]);
+
+  const handleLogout = () => {
+    sessionStorage.removeItem('token');
+    setAuthenticated(false);
+  };
+
+return(
   <ServiceContext.Provider value={defaultServices}>
     <Router>
-      <AppLayout>
-        <AppRoutes />
+      <AppLayout handleLogout={handleLogout}>
+        <AppRoutes authenticated={authenticated} setAuthenticated={setAuthenticated} handleAuthenticated={handleAuthenticated}/>
```

Comment:
Since the `useState` hook `authenticated` was declared in `AppRoutes`, how could I elegantly share that prop with my logout function for`AppLayout`?

Reply:
We would probably want to extract the shared state into something like a `Shared/Services/Login.service.tsx`. Parts of the `Api.service` would be moved into there, too, most likely. This way `AppRoutes` and `AppLayout` can both observe and modify the same pieces of application state, and manage their own small pieces of state for presenting things in their views.

The other alternative is to keep all of this state within a component, and pass down the state pieces (user info, logout function, etc.) as props to both the `AppRoutes` and `AppLayout` component. Both of those are in `index.tsx`, so the top-level `App` component could do this pretty easily.

I think extracting out a service is much cleaner, though.